### ISSUE TITLE
Any suggestions to hide/show a specific column?

### ISSUE THREAD

**Issue by mindcreations** [deitch/jstree-grid#149]
I've tried to play with CSS but I'm not so expert thus I can't figure out how to make it.
I've tried to select elements for the column via jquery to make a .hide() call with no luck.

My need is: in my page I would like to have some buttons to turn on/off some columns.

Please, could you help?

**Comment by deitch** [comment]
Well, there is no built-in API for this. But each column is wrapped in a `div` with a class `jstree-grid-column-<index>`. So if you know your tree root is `id="jstree1"`, you could do:

    $("#jstree1").closest(".jstree-grid-wrapper").find(".jstree-grid-column-2").hide();

It isn't ideal - an API would be better - but it will have to do for now.

**Comment by mindcreations** [comment]
Thank you for your prompt reply :)
In the mean time I had luck with:

$('div[class*="columnName"]').hide()

columnName is currently the _columnClass _ option specified in jstreegrid options. Actually it is only a placeholder for column name, no real style declared.

Of course your 

$("#jstree1").closest(".jstree-grid-wrapper").find(".columnClass").hide(); is far better!

But... an API would be ideal (as you said)... maybe using the column name (aka columnClass or any other new "naming" option) in addition to the column number could be an idea. 
Column number may vary, column name is more stable IMHO.

Maybe you can transform this in a feature request.

**Comment by deitch** [comment]
Try now. Look at the README for the latest release 3.5.16, already pushed out to github and npm.

Lucky it turned out to be so quick and easy. jstree-grid is a side project. It started off at a client, but has not been for a while. I support it when I can, but it usually takes a back door to other work (except when someone actually hires me to work on it, of course).

May I ask, what are you using it for?

**Comment by mindcreations** [comment]
Good :)

Updated to 3.5.16 and used new methods, thanks :)

Il 05/05/2016 06:47, Avi Deitcher ha scritto:
>
> Try now. Look at the README for the latest release 3.5.16, already
> pushed out to github and npm.
>
> Lucky it turned out to be so quick and easy. jstree-grid is a side
> project. It started off at a client, but has not been for a while. I
> support it when I can, but it usually takes a back door to other work
> (except when someone actually hires me to work on it, of course).
>
> May I ask, what are you using it for?
>
Yes, I've built a tree to show and interact with an Optical Distributed
Network hierarchy for our company internal use. Along with tree elements
I've added expand/shrink branch buttons (too much clicks to open a 8
layer hierarchy!), added details on each node, added custom actions with
buttons on an "Action" column of the grid. Also I've placed filter
switches on the top of the grid to switch on/off some columns or to
filter out data. For example we have devices shown in the tree with
ports (imagine a network switch with ports) by default only connected
ports are shown, but if you click "show unused ports button" the tree is
populated also with those along with action button to connect something
to the empty port.

Anyway, thank you for your public donation with jstreegrid :)

Regards,
Stefano

> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/deitch/jstree-grid/issues/149#issuecomment-217076094>
>


-- 
Creator
Mindcreations.com <http://www.mindcreations.com>



---
Questa e-mail è stata controllata per individuare virus con Avast antivirus.
https://www.avast.com/antivirus
